src/openapi/examples.rs

```diff
@@ -780,16 +782,45 @@ fn interesting_params_from_string_type(string: &openapiv3::StringType) -> Vec<se
             .collect();
     }
 
-    // Regex present? Attempt to compile it, and generate a string that matches it
+    // Regex (without anchors) present? Attempt to compile it, and generate a string that matches it
+    let mut contains_anchors = false;
     if let Some(pattern) = &string.pattern {
         match rand_regex::Regex::compile(pattern, 100) {
             Ok(regex_pattern) => {
                 return vec![serde_json::Value::String(
                     regex_pattern.sample(&mut rand::thread_rng()),
                 )]
             }
-            Err(err) => {
-                log::warn!("Broken regex pattern {}, Error message: {}", pattern, err);
+            Err(_err) => {
+                contains_anchors = true;
+            }
+        }
+    }
+
+    // The regex possibly contains anchors (e.g. ^, $)
+    if contains_anchors {
```

Comment:
How about moving this into the Err clause on line 794

Reply:
The two match statements use different patterns. The first one is the original pattern (`pattern`), and the second one (`gen_pattern`) is the pattern with the anchors removed. So to move it into the Err I would have to add another match statement in there, which would lead to very deep nesting

Reply:
I see, @grebnetiew any insightful Rust magic from your end? :)

Reply:
I have used an `if let` to remove a layer of nesting. Also, it seems the existing code returned an error message as the sample string if it could not generate an example that matched the pattern with anchors, which I've changed to a `log::warn`.

Does this meet both of your expectations?

Reply:
LGTM @ringogroenewegen?